examples/animation/animated_transform.rs

```diff
@@ -49,79 +49,79 @@ fn setup(
 
     // Creating the animation
     let mut animation = AnimationClip::default();
-    // A curve can modify a single part of a transform, here the translation
+    // A curve can modify a single part of a transform: here, the translation.
     let planet_animation_target_id = AnimationTargetId::from_name(&planet);
     animation.add_curve_to_target(
         planet_animation_target_id,
-        VariableCurve::linear::<TranslationKeyframes>(
-            [0.0, 1.0, 2.0, 3.0, 4.0],
-            [
-                Vec3::new(1.0, 0.0, 1.0),
-                Vec3::new(-1.0, 0.0, 1.0),
-                Vec3::new(-1.0, 0.0, -1.0),
-                Vec3::new(1.0, 0.0, -1.0),
-                // in case seamless looping is wanted, the last keyframe should
-                // be the same as the first one
-                Vec3::new(1.0, 0.0, 1.0),
-            ],
-        ),
+        UnevenSampleAutoCurve::new([0.0, 1.0, 2.0, 3.0, 4.0].into_iter().zip([
+            Vec3::new(1.0, 0.0, 1.0),
+            Vec3::new(-1.0, 0.0, 1.0),
+            Vec3::new(-1.0, 0.0, -1.0),
+            Vec3::new(1.0, 0.0, -1.0),
+            // in case seamless looping is wanted, the last keyframe should
+            // be the same as the first one
+            Vec3::new(1.0, 0.0, 1.0),
+        ]))
+        .map(TranslationCurve)
+        .expect("Failed to build translation curve"),
```

Comment:
Under what condition could this curve fail to build? If we're passing it static data that we know is good, I feel like the `failed to build translation curve` message is almost misleading. As if this code even has a chance to fail.
```suggestion
        .expect("should be able to build translation curve because we pass in valid samples"),
```